### ISSUE TITLE
Checkbox with children 

### ISSUE THREAD

**Issue by afiorenza** [nvaccess/nvda#5902]
Hi, 
 I have a check boxes and radio buttons with select as children content. When i interact with radio buttons content reads ok, but when interact with check box, when i change the select option and move focus to checkbox input it read the same content. 

[Example fiddle](https://jsfiddle.net/nhsrkrff/1/)

There is an exception that if i turn on browse mode, NVDA reads the content of check box in the right way.

Is this an issue? or any way of fix this with a role, etc?

UPDATE
Found a website that says role="application" should work for strange markup like this https://www.marcozehe.de/2012/02/06/if-you-use-the-wai-aria-role-application-please-do-so-wisely/ . 
It works but i don't know if this role is a viable solution.

Thanks

**Comment by leonardder** [comment]
Ready to paste snippet:

```
data:text/html, <span id="checkbox"><input type="checkbox" aria-labelledby="checkbox--content"><span id="checkbox--content">Option<select name="" id=""><option>A</option>    <option>B</option>    <option>C</option>    </select></span></span>
```

This works as expected in Firefox 54.0.1. Notice that you don't have a label for the select box.

**Comment by jcsteh** [comment]
I don't really understand the issue here. Perhaps you could provide details of exactly what you expected based on your test case (e.g. the message spoken) and what happened instead.

**Comment by Adriani90** [comment]
@afiorenza could you please give some more details on the issue as suggested by @jcsteh in the comment above?
Now, after 1.5 years, is the issue still present?

**Comment by Adriani90** [comment]
Ok I can reproduce this issue in Firefox 65.0.1 in browse mode. I think in the description focus mode and browse mode are confused. In focus mode, NVDA reads the selected option on the checkbox and on the radio button and it also reads the selected option in the combo box correctly. However, in browse mode, NVDA reads the wrong values in the combo box. This is not reproducible in Chrome. @jcsteh your thoughts are appreciated here. Thanks very much.

**Comment by jcsteh** [comment]
The combo box not reading the value is the Firefox bug we're tracking in #7819.

**Comment by Adriani90** [comment]
This is solved in Firefox 71 with NVDA 2019.3 Beta 1.
Closing as works for me.
